packages/components/src/components/CodeEditor.tsx

```diff
@@ -149,41 +151,47 @@ export const CodeEditor = forwardRef<CodeEditorHandle, CodeEditorProps>(
     // Note the `useState` instead of `useMemo`; we never want to recreate EditorState, it would cause bugs.
     // `defaultValue` changes are ignored;
     // `languageSupport` changes (which depends on `project` prop) are handled with `projectRef`.
+
+    const basicExtensions = [
+      highlightSpecialChars(),
+      history(),
+      drawSelection(),
+      dropCursor(),
+      EditorState.allowMultipleSelections.of(true),
+      indentOnInput(),
+      syntaxHighlighting(lightThemeHighlightingStyle, { fallback: true }),
+      bracketMatching(),
+      closeBrackets(),
+      autocompletion(),
+      highlightSelectionMatches({
+        wholeWords: true,
+        highlightWordAroundCursor: false, // Works weird on fractions! 5.3e10K
+      }),
+      compSubmitListener.of([]),
+      compFormatListener.of([]),
+      compViewNodeListener.of([]),
+      keymap.of([
+        ...closeBracketsKeymap,
+        ...defaultKeymap,
+        ...searchKeymap,
+        ...historyKeymap,
+        ...foldKeymap,
+        ...completionKeymap,
+        ...lintKeymap,
+        indentWithTab,
+      ]),
+      compGutter.of([]),
+      compUpdateListener.of([]),
+      compTheme.of([]),
+      languageSupport,
+    ];
+
     const [state] = useState(() =>
       EditorState.create({
         doc: defaultValue,
         extensions: [
-          highlightSpecialChars(),
-          history(),
-          drawSelection(),
-          dropCursor(),
-          EditorState.allowMultipleSelections.of(true),
-          indentOnInput(),
-          syntaxHighlighting(lightThemeHighlightingStyle, { fallback: true }),
-          bracketMatching(),
-          closeBrackets(),
-          autocompletion(),
-          highlightSelectionMatches({
-            wholeWords: true,
-            highlightWordAroundCursor: false, // Works weird on fractions! 5.3e10K
-          }),
-          compSubmitListener.of([]),
-          compFormatListener.of([]),
-          compViewNodeListener.of([]),
-          keymap.of([
-            ...closeBracketsKeymap,
-            ...defaultKeymap,
-            ...searchKeymap,
-            ...historyKeymap,
-            ...foldKeymap,
-            ...completionKeymap,
-            ...lintKeymap,
-            indentWithTab,
-          ]),
-          compGutter.of([]),
-          compUpdateListener.of([]),
-          compTheme.of([]),
-          languageSupport,
+          ...basicExtensions,
+          lineWrapping ? [EditorView.lineWrapping] : [],
```

Comment:
This takes the initial `lineWrapping` prop and won't update it on prop changes.

It's not important in the current playground because settings are in a different tab, but might cause bugs later.

The correct way to do this is:
- create a compartment on top of this file (similar to other `comp*` objects)
- add one more `useEffect` that reconfigures it on `lineWrapping` changes